internal/credentials/docker_test.go

```diff
@@ -7,24 +7,31 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestExtractDockerAuth(t *testing.T) {
+func TestExtractAuthForHost(t *testing.T) {
```

Comment:
You have ExtractAuthConfigs and ExtractBasicAuthForHost but not ExtractAuthForHost.
Either way you are missing tests.

Reply:
Just laziness... I have now done this properly.